### DISCUSSION TITLE
Is YARP Support WCF

### DISCUSSION THREAD

**Discussion by AhmedAssaf** [dotnet/yarp#1327] [Q&A] [answered]
Can YARP Transform from REST to WCF
<img width="635" alt="yarp" src="https://user-images.githubusercontent.com/3175458/138821654-9d0f9760-99b3-409f-8816-981d23302a39.png">

**Comment by Tratcher** [comment, marked as answer]
YARP isn't designed for this kind of translation. YARP is designed to forward an existing request body with updated url and headers. For this scenario you'd have to parse and re-write the body (and more?), possibly in an endpoint specific way.

You'd probably be better off using a standard MVC/API endpoint to receive the REST requests, parse them, and then issue new WCF requests using WCF specific libraries.
